shared/actions/fs/index.js

```diff
@@ -685,6 +685,38 @@ const deleteFile = (state, action: FsGen.DeleteFilePayload) => {
     .catch(makeRetriableErrorHandler(action))
 }
 
+const stateToMoveOrCopyParams = state => ({
+  src: {
+    PathType: RPCTypes.simpleFSPathType.kbfs,
+    kbfs: Constants.fsPathToRpcPathString(state.fs.moveOrCopy.sourceItemPath),
+  },
+  dest: {
+    PathType: RPCTypes.simpleFSPathType.kbfs,
+    kbfs: Constants.fsPathToRpcPathString(
+      Types.pathConcat(
+        state.fs.moveOrCopy.destinationParentPath,
+        Types.getPathName(state.fs.moveOrCopy.sourceItemPath)
+      )
+    ),
+  },
+})
+
+const moveOrCopy = (state, action: FsGen.MovePayload | FsGen.CopyPayload) => {
+  const opID = Constants.makeUUID()
```

Comment:
why not just make the const here and remove the function , aka
```
const moveOrCopy = (state, action: FsGen.MovePayload | FsGen.CopyPayload) => {
  const params = {opID:Constants.makeUUID() , src: { ... /* code form stateToMoveOrCopyParams*/
  ...
  return action.type === FsGen.move ? RPCTypes.SimpleFSSimpleFSMoveRpcPromise(params) : RpcTypes....
```
nothing else will use stateToMoveOrCopyParams right

Reply:
Ah yeah; good catch. I was probalby thinking two approaches at the same time

Reply:
^ above i move opID in there too so you don't need 2 copies